### PR TITLE
place filters on aggrid columns headers

### PR DESCRIPTION
**Pull request by ghelloumi**
Add Aggrid to columns

### CODE REVIEW

File: src/hooks/use-aggrid-sort.js [hunk 1]
```diff
@@ -12,39 +12,44 @@ export const SORT_WAYS = {
     desc: -1,
 };
 
-const getSortSelector = (colKey, sortWay) =>
+const getKeyByValue = (object, value) => {
+    return Object.keys(object).find((key) => object[key] === value);
+};
+
+const getSortSelector = (dataKeyToSortKey, colKey, sortWay) =>
```
Comment (archendel) [inline src/hooks/use-aggrid-sort.js]:
```suggestion
const getSortSelector = (colKey, sortWay) =>
```
unused param

Comment (ghelloumi) [reply src/hooks/use-aggrid-sort.js]: done, thank you

File: src/components/custom-aggrid/custom-aggrid-header.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+    const handleMouseEnter = () => {
+        setDisplayFilterIcon(true);
+    };
+
+    const handleMouseLeave = () => {
+        setDisplayFilterIcon(!!filterSelectedOption || anchorEl);
+    };
+
+    const open = Boolean(anchorEl);
```
[the remark this answers is not in the document]
Comment (ghelloumi) [reply src/components/custom-aggrid/custom-aggrid-header.js]: done, thank you

File: src/components/custom-aggrid/custom-aggrid-header.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    filterOptions,
+    sortConfig,
+    onSortChanged,
+    updateFilter,
+    filterSelectedOption,
+}) => {
+    const { colKey, sortWay } = sortConfig || {};
+    const isSortActive = colKey === field;
+
+    const [anchorEl, setAnchorEl] = useState(null);
```
[the remark this answers is not in the document]
Comment (ghelloumi) [reply src/components/custom-aggrid/custom-aggrid-header.js]: done, thank you

File: src/components/custom-aggrid/custom-aggrid-header.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+    const handleShowFilter = (event) => {
+        setFilterAnchorEl(event.currentTarget);
+    };
+
+    const handleCloseFilter = () => {
+        setFilterAnchorEl(null);
+        if (!filterSelectedOption) {
+            setIsHoveringHeader(false);
+        }
```
Comment (archendel) [inline src/components/custom-aggrid/custom-aggrid-header.js:47]:
```suggestion
        setIsHoveringHeader(false);
```

**Review by archendel — APPROVED** [review]
Last small change then code and tests OK

File: src/components/custom-aggrid/custom-aggrid-header.js (hunk 4 above)
Comment (ghelloumi) [reply src/components/custom-aggrid/custom-aggrid-header.js]: done, thank you

**Pull request merged**